### PR TITLE
Higher order functions

### PR DESCRIPTION
**Pull request by crodeheaver**
Should be all there

### CODE REVIEW

File: zoo.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+displayFeaturedAnimal(featured[0]);
+
+var ages = animals.map(picks);
+
+ages.sort(sortByAge);
+displayAnimalAges(ages)
+
+
+function sortBy(prop) {
+    if (prop === "age") {
```
Comment (barberboy) [inline zoo.js]: This should be much more generic, i.e. don't refer specifically to "age" or "name" in the function.

Maybe by default it just does `return a < b ? -1 : a > b ? 1 : 0;` but change the signature to accept a second parameter, `ignoreCase`, that you use like `var sortByName = sortBy("name", true);`

Does that make sense?

**Comment by crodeheaver** [comment]
so something like this?

File: zoo.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+
+function sortBy(prop, ignoreCase) {
+    if(!ignoreCase)
+        return function(a, b) {
+            return a[prop] < b[prop] ? -1 : a[prop] > b[prop] ? 1 : 0;
+        }
+    else
+        return function(a, b) {
+            return function(a, b) {
```
Comment (barberboy) [inline zoo.js]: One too many nested functions.

**Comment by crodeheaver** [comment]
Woops. I was in a bit of a rush

**Comment by barberboy** [comment]
👍🏻💯

**Pull request closed without merging**